src/controllers/puppeteer-controller.js

```diff
@@ -3,6 +3,61 @@ export class PuppeteerController {
         this.service = puppeteerService;
     }
 
+    async scrape(req, res) {
+        try {
+            if (!req.body.url) {
+                res.status(400).send('The url field is rquired');
+                return;
+            }
+            if (!req.body.selectors && !Array.isArray(req.body.selectors)) {
+                res.status(400).send('The selectors field is rquired and must be an array of objects');
+                return;
+            }
+            for (const s of req.body.selectors) {
+                console.log(s);
+                if (!s.name) {
+                    res.status(400).send('Selectors require the field: name');
+                    return;
+                }
+                if (!s.selector) {
+                    res.status(400).send('Selectors require the field: selector');
+                    return;
+                }
+            }
+
+            const results = await this.service.eval(req.body.url, req.body.selectors, sels => {
+                return sels.map(sel => {
+                    sel.results = Array.from(document.querySelectorAll(sel.selector)).map(x => {
+                        switch (sel.type) {
+                            case 'link':
+                                return {
+                                    content: x.innerText.trim(),
+                                    link: x.getAttribute('href')
+                                };
+                            case '':
```

Comment:
Is there a point to `case '':`? seems redundant.

Reply:
was supposed to be 'html'